src/dom.ts

```diff
@@ -269,3 +269,67 @@ export function compareElementArea(a: HTMLElement, b: HTMLElement): number {
 
     return aArea - bArea
 }
+
+/** Computes a unique CSS selector for the current HTMLElement.
+ *  Currently buggy, returns a unique selector but doesn't always correspond
+ *  to the selector you should use for functions such as querySelectorAll().
+ */
+export function getSelector(elem: HTMLElement) {
+   if (elem === null || elem === undefined)
+      return undefined
+
+   if (elem == document.body || !elem.parentNode)
+      return "body"
+
+   if (elem.id)
+      return "#" + elem.id
+
+   let parent = elem.parentNode as HTMLElement
+   let n = [].indexOf.call(parent.childNodes, elem) + 1
+   return getSelector(parent) + " :nth-child(" + n + ")"
+}
+
+export const hintworthy_js_elems = {}
+
+/** Adds or removes an element from the hintworthy_js_elems array of the
+ *  current tab.
+ *  This function is made available to the page's. We should be very careful
+ *  about the input it accepts and what it does with it.
+ */
+export function registerEvListenerAction(action: string, ev: string): void {
+   if (!(this instanceof HTMLElement))
+      return
+
+   switch (ev) {
+      case "click":
+      case "hover":
+      case "mouseover":
+         if (action == "addEventListener")
+            hintworthy_js_elems[getSelector(this)] = this
+         else
+            delete hintworthy_js_elems[getSelector(this)]
+   }
+}
+
+/** Replace the page's addEventListener with a closure containing a reference
+ *  to the original addEventListener and our registering function. Do the
+ *  same with removeEventListener.
+ */
+export function hijackPageListenerFunctions(): void {
+   let exportedName = 'registerEvListenerAction'
+   exportFunction(registerEvListenerAction, window, {defineAs: exportedName})
+
+   let eval_str = ["addEventListener", "removeEventListener"].reduce((acc, cur) => `${acc};
+      EventTarget.prototype.${cur} = ((realFunction, register) => {
+         console.log("Replacing page's ${cur} with Tridactyl's.")
+         return function () {
+               // console.log("Calling Tridactyl's proxy for the page's ${cur}.")
+               let args = [].slice.call(EventTarget.prototype.${cur}.arguments)
```

Comment:
I thought the `...args` syntax was typescript sugar, I didn't know it was available in pure Javascript. We need to keep the `apply` for realFunction though, by removing them we would pass the `args` array to addEventListener instead of putting each arg where it belongs.